### PR TITLE
Source: Amazon Ads - Improve error handling for attribution_report(s) streams

### PR DESCRIPTION
**Pull request by grubberr**
## What
Try to improve oncall https://github.com/airbytehq/oncall/issues/1955

- re-implement AttributionReport using slices
- improve error handling for 400, "This profileID is not authorized to use Amazon Attribution"
- remove "strange" unit_test test_attribution_report_no_data

Amazon Doc: https://advertising.amazon.com/API/docs/en-us/amazon-attribution-prod-3p/#/Reports/getAttributionTagsByCampaign

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Review by artem1205 — APPROVED** [review]

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py [hunk 1]
```diff
@@ -70,40 +72,51 @@ class AttributionReport(AmazonAdsStream):
...
         return "POST"
 
-    def read_records(self, *args, **kvargs) -> Iterable[Mapping[str, Any]]:
-        """
-        Iterate through self._profiles list and send read all records for each profile.
-        """
+    def path(self, **kvargs) -> str:
+        return "/attribution/report"
+
+    def stream_slices(
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py:84]: do you test new functionality with unit tests?

Comment (grubberr) [reply airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/attribution_report.py:84]: I have added unit_test for slicing

**Review by lazebnyi — APPROVED** [review]
Nice!

**Pull request merged**